File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        best_time_to_buy = prices[0]
        
        ans = 0
        
        for i in range(1, len(prices)):
            if prices[i] < best_time_to_buy:
                best_time_to_buy = prices[i]
            else:
                ans = max(ans, prices[i] - best_time_to_buy)
        
        return ans

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
#include <algorithm>

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int ans = 0;
        int buy_price = prices[0];
        
        for(int i=1; i < prices.size(); i++){
            if (prices[i] <= buy_price){
                buy_price = prices[i];
            } else{
                ans = max(ans, prices[i] - buy_price);
            }
        }
        
        return ans;
        
    }
};